src/Service/ShelfManager.php

```diff
@@ -23,18 +19,19 @@ public function getBooksInShelf(Shelf $shelf): array
             return $shelf->getBooks()->toArray();
         }
 
-        $qb = $this->entityManager->getRepository(Book::class)->getAllBooksQueryBuilder();
-
-        $request = new Request($shelf->getQueryString());
-
-        $this->createAndHandleFilter(BookFilterType::class, $qb, $request);
-
-        $results = $qb->getQuery()->getResult();
-
-        if (!is_array($results)) {
-            return [];
+        $tokens = $this->queryTokenizer->tokenize($shelf->getQueryString());
+        $complexQuery = $this->tokenHandler->handle($tokens);
```

Comment:
`handle` is too generic,  Use another name, like `$this->tokensToTypesenseQueryAdapter->getQuery($tokens)`